src/TaylorN.jl

```diff
@@ -585,6 +626,9 @@ function square(a::HomogeneousPolynomial)
     return HomogeneousPolynomial{T}(coeffs, order)
 end
 
+square(a::HomogeneousPolynomial) = a*a
```

Comment:
I think this line is part of the problem; the function defined on line 599 on is optimized (roughly a factor 2 in comparison with a*a). While the differences are not too bad,, I'm not quite getting the results without `mul!`...